linux/scripts/upload-to-debian.sh

```diff
@@ -73,57 +79,75 @@ if ! git diff --quiet; then
     builder_die "You have changed files in your git working directory. Exiting."
 fi
 
+function get_latest_stable_branch_name() {
+  # Get the list of stable branches and take the one with the highest number.
+  # Extract `origin/stable-16.0` from `  origin/stable-16.0`
+  local branches stable_branch
+  branches=$(git branch -r | grep origin/stable-)
+  stable_branch=$(echo "${branches}" | sort | tail -1)
+  if [[ -z ${stable_branch} ]]; then
+    builder_die "Can't find stable-* branch" > /dev/stderr
```

Comment:
Will this fail the script, given it is called in a subshell on line 120?

Reply:
Yes, it will work as intended: If one of the commands in the subshell fails we don't end the subshell (because `set -e` is not set in the subshell), but we check that we found a stable branch and return that. If we didn't find a stable branch (either because there is none, or because one of the commands to retrieve it failed) we exit the subshell with an errorcode and that exits the script (because the script itself runs with `set -e`).